src/components/Banner/Banner.tsx

```diff
@@ -0,0 +1,252 @@
+  "dark.section.whats-on.primary",
+] as const;
+export type BannerHighlightColors = typeof bannerHighlightColorsArray[number];
+
+interface BannerProps {
+  /** Label used to describe the `Banner`'s aside HTML element. */
+  ariaLabel?: string;
+  /** Used to set the color of the background for the full component.
+   * Refer to how color values are defined and typed in the DS Icon component. */
+  backgroundColor?: BannerBgColors;
+  /** Additional `className` to add. */
+  className?: string;
+  /** Used to populate the body content of the component. */
+  content: string | JSX.Element;
+  /** Used to populate the heading element within the component.  A string
+   * value can be passed to set the text for a default DS Heading component,
+   * or an actual DS Heading component can be passed in. */
+  heading?: string | JSX.Element;
+  /** Used to set the color for the left border and icon.  Refer to how color
+   * values are defined and typed in the DS Icon component. */
+  highlightColor?: BannerHighlightColors;
+  /** Optional custom `Icon` that will override the default `Icon`. */
+  icon?: JSX.Element;
+  /** ID that other components can cross reference for accessibility purposes. */
+  id?: string;
+  /** Optional prop to control whether a `Banner` can be dismissed
+   * (closed) by a user. */
+  isDismissible?: boolean;
+  /** Used to control the component's semantic coloring and iconography. */
+  type?: BannerTypes;
+}
+
+const iconProps: Record<BannerTypes, IconProps> = {
+  neutral: {
+    name: "errorOutline",
+    title: "Banner neutral icon",
+    iconRotation: "rotate180",
+  },
+  informative: {
+    name: "errorOutline",
+    title: "Banner informative icon",
+    iconRotation: "rotate180",
+  },
+  positive: {
+    name: "actionCheckCircle",
+    title: "Banner positive icon",
+  },
+  negative: {
+    name: "errorOutline",
+    title: "Banner negative icon",
+  },
+  warning: {
+    name: "alertNotificationImportant",
+    title: "Banner warning icon",
+  },
+  recommendation: {
+    name: "actionLightbulb",
+    title: "Banner recommendation icon",
+  },
+};
+
+/**
+ * The `Banner` component is a non-modal semantic dialog used to communicate a
+ * general status event or to promote a feature, providing contextual feedback
+ * messages for typical user actions. They are displayed contextually within a
+ * page flow and they will often prompt a user to take action.
+ */
+export const Banner: ChakraComponent<
+  React.ForwardRefExoticComponent<
+    BannerProps & React.RefAttributes<HTMLDivElement>
+  >,
+  BannerProps
+> = chakra(
+  forwardRef<HTMLDivElement, BannerProps>((props, ref?) => {
+    const {
+      ariaLabel,
+      backgroundColor,
+      className,
+      content,
+      heading,
+      highlightColor,
+      icon,
+      id,
+      isDismissible = false,
+      type = "neutral",
+      ...rest
+    } = props;
+    const [isOpen, setIsOpen] = useState(true);
+    const handleClose = () => setIsOpen(false);
```

Comment:
Should this be opened up to consuming apps so they can keep track of banners that have been closed?

Reply:
But it would mean passing in an additional `onClose` prop. Perhaps for a v2?

Reply:
Yes.  https://jira.nypl.org/browse/DSD-1756